stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2511,6 +2511,7 @@ impl NakamotoChainState {
                 Self::calculate_matured_miner_rewards(
                     &mut clarity_tx,
                     sortition_dbconn.sqlite_conn(),
+                    // coinbase_height + 1,
```

Comment:
@kantai I had to comment this out because it was causing the chain to crash in some of the integration tests.  Can you please help me understand why this change was here in the first place?

Reply:
There were some tests where the coordinator panicked without this change (I think they were signers tests that used mockamoto, or some configuration where nakamoto was active during very early Stacks block heights). This was because of some assumptions about the block height, maturation windows, and the size of the matured rewards. I don't think `+1` here is correct, but I do think that the height used for calculated miner rewards maturation in nakamoto needs another look.

The `coinbase_height` parameter of `NakamotoChainState::setup_block` includes the block being set up. The `chain_tip` parameter of `StacksChainState::setup_block`, however, is the parent tip, so that the stacks height used in this function is logically `-1`. This is why `StacksChainState` uses `+1` when invoking `handle_prepare_phase_start()`. This was discovered in trying to test that method, but I didn't try to completely test all of the behavior around these height parameters. It may be that the first block of a tenure has the parent's height passed in, but subsequent blocks in that tenure have the new coinbase height (where we could see something like https://github.com/stacks-network/stacks-core/issues/4333).